mysqld.sh

```diff
@@ -355,5 +355,5 @@ fi
 # Start mysqld
 echo "${LOG_MESSAGE} ---------------------------------------------------------------"
 echo "${LOG_MESSAGE} Starting with options: $OPT $START"
-exec mysqld $OPT $START
+exec mysqld $OPT $START --bind-address=0.0.0.0
```

Comment:
Not sure why this is needed, but I think it belongs in one of the .cnf files instead.

Reply:
It is already here: https://github.com/colinmollenhour/mariadb-galera-swarm/blob/master/conf.d/30-galera.cnf#L2

Are you overwriting this file locally, perhaps?

Reply:
The issue I guess is that, at least in the jsurf image and linuxserver image, the bind-address setting gets overwritten by /etc/mysql/mariadb.conf.d/50-server.cnf file, with 127.0.0.1.

The breakdown:
`mysqladmin --help` command lists the following files
```
Default options are read from the following files in the given order:
/etc/my.cnf /etc/mysql/my.cnf ~/.my.cnf
```
Out of those, only the `/etc/mysql/my.cnf` file exists, with the following ending:
```
# Import all .cnf files from configuration directory
!includedir /etc/mysql/conf.d/
!includedir /etc/mysql/mariadb.conf.d/
```
The first directory is the one the dockerfile populates, the second is presumably mariadb default.

Grepping for the bind-address yields the following output:
```
root@5d24286d9f20:/etc/mysql# grep -r "bind-address"
conf.d/30-galera.cnf:bind-address=0.0.0.0
mariadb.conf.d/50-server.cnf:bind-address               = 127.0.0.1
```
As the `/etc/mysql/mariadb.conf.d/` gets included later, I believe it overwrites the `/etc/mysql/conf.d/` settings.

I guess we should then overwrite the `mariadb.conf.d/50-server.cnf` through the dockerfile instead of my command argument solution.

Reply:
Yeah I'd like to avoid making major changes to the config to avoid impacting existing users, so probably just resetting the environment from the jsurf image to match that of the official image would be the best soluition. E.g. `RUN rm /etc/mysql/conf.d/*` or similar.